### PR TITLE
feat(client): ✨ Create view Account page and functionality

### LINKED ISSUE #41: Catniss can view her account page
From the splash page, Catniss can (if she is logged in) click the account button to view her account page so that she can edit various account settings.

This page includes a password change feature and the ability to sign out in case she wants to change her current password or she does not want the account to stay signed in for any reason.

note that the account page is different from the stats page. the stats page is #43 

Tasks

- [x] Show Catniss email on page
- [x] allow Catniss to change password if signed in through email
- [x] Allow Catniss to sign out through account page
- [x] Make as a page

### Acceptance criteria
- user can click account button and be taken to the account page (should be a real `<a>` with a valid `href`)
- account page has working sign out button
- user can change password on account page
- if use is signed in with google or microsft they should be able to add a password to there account
- if the user is not signed in using google they should be able to link a google signin to there account
- if the user is not signed in using microsoft they should be able to link a google microsoft to there account

### PR DESCRIPTION
**Pull request by zaverbac**

### CODE REVIEW

File: client/src/routes/settings/+page.svelte
Comment (gchamb) [inline client/src/routes/settings/+page.svelte]: This displays I successfully linked google account even though i only clicked the button.

File: client/src/routes/settings/+page.svelte
Comment (gchamb) [inline client/src/routes/settings/+page.svelte]: This displays I successfully linked microsoft account even though i only clicked the button.

File: client/src/lib/firebase/auth.ts
Comment (gchamb) [inline client/src/lib/firebase/auth.ts]: probably want to return `linkWithPopup` then awaited when you use it

File: client/src/lib/firebase/auth.ts
Comment (gchamb) [inline client/src/lib/firebase/auth.ts]: probably want to return linkWithPopup then awaited when you use it

File: client/src/routes/settings/+page.svelte
Comment (gchamb) [inline client/src/routes/settings/+page.svelte]: Dialog doesn't close after you link the google/microsoft acc

File: client/src/routes/settings/+page.svelte
Comment (gchamb) [inline client/src/routes/settings/+page.svelte]: probably want to use authStore to get the users email because it isn't updating after i link my google acc

File: client/src/routes/settings/+page.svelte
Comment (gchamb) [inline client/src/routes/settings/+page.svelte]: works but doesn't closes the dialog after setting the password

File: client/src/lib/firebase/auth.ts
Comment (gchamb) [reply client/src/lib/firebase/auth.ts:64]: probably want to return `linkWithPopup` then awaited when you use it

**Comment by gchamb** [comment]
On line 114 in auth.ts where is getDisplayName function being used for?

**Review by gchamb — DISMISSED** [review]
see changes above

File: client/src/routes/settings/+page.svelte
Comment (gchamb) [inline client/src/routes/settings/+page.svelte]: you dont need to return

File: client/src/routes/settings/+page.svelte
Comment (gchamb) [inline client/src/routes/settings/+page.svelte]: no need to return

File: client/src/routes/settings/+page.svelte
Comment (gchamb) [inline client/src/routes/settings/+page.svelte]: no need to return

File: client/src/routes/settings/+page.svelte
Comment (gchamb) [inline client/src/routes/settings/+page.svelte]: no need to return

File: client/src/routes/settings/+page.svelte
Comment (gchamb) [inline client/src/routes/settings/+page.svelte]: no need to return

**Comment by gchamb** [comment]
refresh page so updated link google/microsoft acc can update the email field

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts:16]: lets do these inside of a function

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: I feel like this should be an error

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: I feel like this should be an error

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: I feel like this should be an error

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: lets do these inside of a function

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: what?

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: dont need this because we have a show password button

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: lets show the error message instead of just saying unknown

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: this button should only be here if you don't alread have google account linked

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: his button should only be here if you don't alread have microsft account linked

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: I would prefer if this function had an argument and a return instead of mutating the state

**Review by nstringham — CHANGES_REQUESTED** [review]

File: client/src/lib/firebase/auth.ts
Comment (njumaah) [inline client/src/lib/firebase/auth.ts]: hasGoogleProvider is a better naming convention

File: client/src/lib/firebase/auth.ts
Comment (njumaah) [inline client/src/lib/firebase/auth.ts]: hasMicrosoftProvider is a better naming convention

File: client/src/routes/settings/+page.svelte
Comment (njumaah) [inline client/src/routes/settings/+page.svelte]: duplicate code from signinbutton component. make another component and insert them in each

File: client/src/routes/settings/+page.svelte
Comment (njumaah) [inline client/src/routes/settings/+page.svelte]: make a more consistent user message

**Review by njumaah — CHANGES_REQUESTED** [review]

**Comment by njumaah** [comment]
make to to pull from dev, then let me review again before merging

**Review by njumaah — DISMISSED** [review]
lgtm, wait for nates review before merging

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: this would probobly be better as a guard (aka early return)

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: this would probobly be better as a guard (aka early return)

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: I would rather we just return the message in the else case

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: let's give this a more descriptive name

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: these could just be returned from the function instead of being transformed

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: you are not doing anything with the error message

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: you are not doing anything with the error message

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: you are not doing anything with the error message

File: client/src/routes/+page.svelte
Comment (nstringham) [inline client/src/routes/+page.svelte]: I don't think you want this

File: client/src/components/ProviderButtons.svelte
Comment (nstringham) [inline client/src/components/ProviderButtons.svelte]: these styles are duplicated in the settings page and in the account component

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: this should do a page reload like it does for the sign out button in the account dropdown

**Review by nstringham — DISMISSED** [review]

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: you put a blank line here in `hasGoogleProvider` but not in `hasMicrosoftProvider`

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte:104]: `googleErr`, `microsoftErr`, and `deleteErr` are never displayed

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: this should do a reload like logout does

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [reply client/src/lib/firebase/auth.ts]: fixed

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [reply client/src/lib/firebase/auth.ts]: fixed

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: consider using `??`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: you also don't need this `return`

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [reply client/src/routes/settings/+page.svelte]: still an issue

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [reply client/src/routes/settings/+page.svelte]: still an issue

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [reply client/src/routes/settings/+page.svelte]: still an issue

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [reply client/src/routes/settings/+page.svelte]: still an issue

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [reply client/src/lib/firebase/auth.ts]: fixed

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [reply client/src/lib/firebase/auth.ts]: fixed

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [reply client/src/routes/settings/+page.svelte]: @njumaah is this resolved now? if not can you give an example of what you think would be a better error message or explain what is inconsistent about the current message.

Comment (zaverbac) [reply client/src/routes/settings/+page.svelte]: I believe he was just saying be consistent with the wording of sign in as I was saying SignIn on one part and sign in on the other

File: client/src/components/AccountButton.svelte
Comment (nstringham) [inline client/src/components/AccountButton.svelte]: we still need the email button code

File: client/src/components/ProviderButtons.svelte
Comment (nstringham) [inline client/src/components/ProviderButtons.svelte]: there are no email stuff in this component

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: you don't need to relaod here because logout will already reload

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [reply client/src/routes/settings/+page.svelte]: I think this is done now

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [reply client/src/routes/settings/+page.svelte]: I think this is done now

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [reply client/src/routes/settings/+page.svelte]: I think this is done now

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [reply client/src/routes/settings/+page.svelte]: I think this is done now

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [reply client/src/routes/settings/+page.svelte]: I think this is done now

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: this button should be an action not in content

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: I don't think you want to export this

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [reply client/src/routes/settings/+page.svelte]: resolved in 3ee8bb956cde4dc4ea1245bf9c6cb09f60c1b3b2

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: this needs to be awaited or returned

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: this doesn't work because you need to await deleteAccount

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [reply client/src/lib/firebase/auth.ts]: fixed in f7acc7b53c359a49a8d55d432d7edb0e5a160dee

File: client/src/routes/settings/+page.svelte
Comment (zaverbac) [reply client/src/routes/settings/+page.svelte]: Fixed

File: client/src/routes/settings/+page.svelte
Comment (zaverbac) [reply client/src/routes/settings/+page.svelte]: Using authStore to get email

File: client/src/routes/settings/+page.svelte
Comment (zaverbac) [reply client/src/routes/settings/+page.svelte]: Closes dialog

File: client/src/routes/settings/+page.svelte
Comment (zaverbac) [reply client/src/routes/settings/+page.svelte]: Added await to both functions

File: client/src/lib/firebase/auth.ts
Comment (zaverbac) [reply client/src/lib/firebase/auth.ts]: awaited function

File: client/src/routes/settings/+page.svelte
Comment (zaverbac) [reply client/src/routes/settings/+page.svelte]: removed export

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: if you are checking this in the if that throws an error you don't need to check it here

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: if you are checking this in the if that throws an error you don't need to check it here

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: if you are checking this in the if that throws an error you don't need to check it here

**Review by nstringham — CHANGES_REQUESTED** [review]

File: client/src/routes/settings/+page.svelte
Comment (nstringham) [inline client/src/routes/settings/+page.svelte]: what does this if statment do?

Comment (zaverbac) [reply client/src/routes/settings/+page.svelte]: it opens a dialog that tells the user that they need to sign in again and they click ok and it redirects them back to the sign in  page and logs them out

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: you can remove `user === undefined ||` because it is not doing anything

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: you can remove `user === undefined ||` because it is not doing anything

File: client/src/lib/firebase/auth.ts
Comment (nstringham) [inline client/src/lib/firebase/auth.ts]: you can remove `user === undefined ||` because it is not doing anything

File: client/src/routes/+page.svelte
Comment (nstringham) [inline client/src/routes/+page.svelte]: this seams like a merge conflict thing I don't think you are meaning to do this

**Review by nstringham — CHANGES_REQUESTED** [review]

File: client/src/routes/+page.svelte
Comment (nstringham) [inline client/src/routes/+page.svelte]: Zac, why does this remove every single empty line in this file?

**Review by nstringham — CHANGES_REQUESTED** [review]

File: client/src/routes/+page.svelte
Comment (zaverbac) [reply client/src/routes/+page.svelte]: I'm not sure. I didn't remove anything. I know it wasn't showing the splash page on the merge page so I merged it manually by grabbing this from dev and replacing the one I had and changing the SignIn component to AccountButton

Comment (zaverbac) [reply client/src/routes/+page.svelte]: Fixed

**Review by nstringham — APPROVED** [review]

**Review by njumaah — APPROVED** [review]
All acceptance criteria has been taken care of.

**Pull request merged**
